File: testdata/jennies/builders/discriminator_without_option/JavaBuilder/discriminator_without_option/NoShowFieldOptionBuilder.java
package discriminator_without_option;


public class NoShowFieldOptionBuilder implements cog.Builder<NoShowFieldOption> {
    protected final NoShowFieldOption internal;
    
    public NoShowFieldOptionBuilder() {
        this.internal = new NoShowFieldOption();
    }
    public NoShowFieldOptionBuilder text(String text) {
        this.internal.text = text;
        return this;
    }
    public NoShowFieldOption build() {
        return this.internal;
    }
}

File: testdata/jennies/builders/discriminator_without_option/PHPBuilder/src/DiscriminatorWithoutOption/NoShowFieldOptionBuilder.php
<?php

namespace Grafana\Foundation\DiscriminatorWithoutOption;

/**
 * @implements \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\DiscriminatorWithoutOption\NoShowFieldOption>
 */
class NoShowFieldOptionBuilder implements \Grafana\Foundation\Cog\Builder
{
    protected \Grafana\Foundation\DiscriminatorWithoutOption\NoShowFieldOption $internal;

    public function __construct()
    {
    	$this->internal = new \Grafana\Foundation\DiscriminatorWithoutOption\NoShowFieldOption();
    }

    /**
     * Builds the object.
     * @return \Grafana\Foundation\DiscriminatorWithoutOption\NoShowFieldOption
     */
    public function build()
    {
        return $this->internal;
    }

    public function text(string $text): static
    {
        $this->internal->text = $text;
    
        return $this;
    }

}
